_pytest/main.py

```diff
@@ -100,11 +100,7 @@ def wrap_session(config, doit):
             session.exitstatus = EXIT_INTERNALERROR
             if excinfo.errisinstance(SystemExit):
                 sys.stderr.write("mainloop: caught Spurious SystemExit!\n")
-        else:
-            if session.testsfailed:
-                session.exitstatus = EXIT_TESTSFAILED
-            elif session.testscollected == 0:
-                session.exitstatus = EXIT_NOTESTSCOLLECTED
+
```

Comment:
no, i think we should move it to mainline before - @nicoddemus did the original refactoring which i had to partially revert/change to have cache output good return codes for commands unrelated to running tests